src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -246,15 +246,18 @@ def arguments_loader():
     def handler(args):  # pylint: disable=too-many-branches,too-many-statements
         from msrestazure.azure_operation import AzureOperationPoller
 
-        ordered_arguments = args.pop('ordered_arguments') if 'ordered_arguments' in args else []
+        ordered_arguments = args.pop('ordered_arguments', [])
+        for item in ['properties_to_add', 'properties_to_set', 'properties_to_remove']:
+            if args[item]:
+                raise CLIError("Unexpected '{}' was not empty.".format(item))
```

Comment:
Shall we just say `'{}' was not expected'`. I don't know much about generic update though, but in case it helps
Also why we want to error out here? I assume we will apply their values on sending out the `update` payload?